authentication.js

```diff
@@ -12,24 +12,9 @@ zapier.tools.env.inject();
  * @returns {Promise<T | never>}
  */
 const test = (z, bundle) => {
-  z.request({
+  return z.request({
     url: `${process.env.PLATFORM_URL}/api/permission/systemuser/profile/`,
-    headers: {
-      Authorization: `Token ${bundle.authData.email}:${bundle.authData.token}`,
-    },
   }).then((response) => {
-    if (response.status === 400) {
-      z.console.log('A bad request status happened while trying the URL');
-      throw new Error('400 Bad request');
-    }
-    else if (response.status === 401) {
-      z.console.log('Oops, got a 401');
-      throw new Error('The username and/or password you supplied is incorrect');
-    } else if (response.status === 403) {
-      z.console.log('This url is forbidden');
-      throw new Error('You are not allowed to access this URL');
-    }
```

Comment:
What about other exceptions, are there any which can happen? And should we in those cases also throw an exception or can we ignore it? If we need to throw other error's also maybe then do the same you do in `index.js` and check for `if (response.status === 200)` first and then add an `else` branch for the generic errors and log also the error number for more info.

Reply:
`index.js => processResponseCode()` is a middleware function that gets called on every request, so since this function does a `z.request()` it first calls `addAuthorizationHeader` then does the request and then calls `processResponseCode()`.
Thus it already checks for a `200`, `401`, and `403`.  If there are any other response codes returned it will log that, and we can see if there needs to be an exception.